packages/core/src/compiler/Domain.test.ts

```diff
@@ -168,13 +168,13 @@ type Set
   });
   test("Type not found", () => {
     const prog = `
-constructor Cons ['X] : 'X head * List('X) tail -> List('X)
+constructor Cons ['X] ('X head, List('X) tail) -> List('X)
```

Comment:
Now that I think about it, perhaps a more conventional `constructor Cons<'X> ('X head, List<'X> tail) -> List<'X>` is more readable?

Reply:
Also funny that everything is just looking more like typescript 😛

Reply:
> Now that I think about it, perhaps a more conventional `constructor Cons<'X> ('X head, List<'X> tail) -> List<'X>` is more readable?

Yeah, I'd like that a lot better too I think; should I make an issue about this so it can be done in a followup PR?

Reply:
Sounds good!

Reply:
Done: #742